### PR TITLE
events integration tests

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: server/event_integration_test.go [hunk 1]
```diff
@@ -45,6 +52,346 @@ func (s *EventIntegrationTestSuite) TearDownTest() {
...
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+}
+
+func (s *EventIntegrationTestSuite) Test_CreateAppEvent_App_is_disabled() {
+	appID := uuid.NewString()
+	expectedStatusCode := http.StatusBadRequest
```
Comment (Subomi) [inline server/event_integration_test.go:108]: This is a bug, I created an issue for it here #650

File: server/event_integration_test.go [hunk 2]
```diff
@@ -45,6 +52,346 @@ func (s *EventIntegrationTestSuite) TearDownTest() {
...
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep Assert.
+	var respEvent datastore.Event
+	parseResponse(s.T(), w.Result(), &respEvent)
+	event.DocumentStatus = ""
+	event.ID = primitive.ObjectID{}
+	require.Equal(s.T(), *event, respEvent)
```
Comment (Subomi) [inline server/event_integration_test.go:150]: Compare IDs.

File: server/event_integration_test.go [hunk 3]
```diff
@@ -45,6 +52,346 @@ func (s *EventIntegrationTestSuite) TearDownTest() {
...
+	parseResponse(s.T(), w.Result(), &respEventDelivery)
+	eventDelivery.ID = primitive.ObjectID{}
+	eventDelivery.DocumentStatus = ""
+	require.Equal(s.T(), datastore.ScheduledEventStatus, respEventDelivery.Status)
+
+	// clear both delivery statuses
+	eventDelivery.Status = ""
+	respEventDelivery.Status = ""
+
+	require.Equal(s.T(), *eventDelivery, respEventDelivery)
```
Comment (Subomi) [inline server/event_integration_test.go:235]: Compare IDs. Forget clearing status.

**Review by Subomi — CHANGES_REQUESTED** [review]
@danvixent FYI

File: server/event_integration_test.go [hunk 4]
```diff
@@ -45,6 +52,346 @@ func (s *EventIntegrationTestSuite) TearDownTest() {
...
+	}
+
+	require.Equal(s.T(), v, respEvents)
+}
+
 func TestEventIntegrationSuiteTest(t *testing.T) {
 	suite.Run(t, new(EventIntegrationTestSuite))
 }
+
+func newRequestAndResponder(method string, url string, body io.Reader) (*http.Request, *httptest.ResponseRecorder) {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply server/event_integration_test.go:383]: 1. The CI linter keeps flagging them as unused code, not sure why, so i moved it here temporarily.
2. okay
3. I'm aware, but in my opinion, this block of code is not complex enough but it is repeated several times throughout the tests that it warrants a small helper.

File: server/event_integration_test.go [hunk 5]
```diff
@@ -45,6 +52,346 @@ func (s *EventIntegrationTestSuite) TearDownTest() {
...
+	req := httptest.NewRequest(method, url, body)
+	req.SetBasicAuth("test", "test")
+	req.Header.Add("Content-Type", "application/json")
+
+	return req, httptest.NewRecorder()
+}
+
+func serialize(t *testing.T, obj interface{}) io.Reader {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply server/event_integration_test.go:391]: > 1. The CI linter keeps flagging them as unused code, not sure why, so i moved it here temporarily.
Noted the rest.

**Review by jirevwe — APPROVED** [review]

**Pull request merged**